types/node/buffer.buffer.d.ts

```diff
@@ -160,6 +160,26 @@ declare module "buffer" {
              * @param encoding The encoding of `string`. **Default:** `'utf8'`.
              */
             from(string: WithImplicitCoercion<string>, encoding?: BufferEncoding): Buffer<ArrayBuffer>;
+            /**
+             * Creates a view of the `ArrayBuffer` or a new `Buffer` containing `string`.
+             * The `encoding` parameter identifies the character encoding to be used when
+             * converting `string` into bytes.
+             *
+             * A `TypeError` will be thrown if `string` is not a string or another type
+             * appropriate for `Buffer.from()` variants.
+             *
+             * `Buffer.from(string)` may also use the internal `Buffer` pool like
+             * `Buffer.allocUnsafe()` does.
+             * @since v5.10.0
+             * @param arrayBufferOrString An array buffer or a string to encode.
+             * @param offsetOrEncoding A byte-offset or encoding.
+             * @param length A length.
+             */
+            from(
+                arrayBufferOrString: WithImplicitCoercion<string | ArrayBufferLike>,
+                offsetOrEncoding?: number | BufferEncoding,
+                length?: number,
+            ): Buffer<ArrayBufferLike>;
```

Comment:
The issue with this signature is that it violates the TS5.7+ behaviour for typed array constructors – that is, that passing a `T extends ArrayBufferLike` to the constructor should result in a `TypedArray<T>`. This was a thought when patching, but the reality is that these are different signatures with different parameters.

I don't know whether @rbuckton has any thoughts. My $0.02 would be:

- merging the signatures is less than ideal for the above reasons
- if there were to be a combined signature, it should only accept the single source parameter, and not accept additional parameters which are divergent between signatures

Reply:
I can try narrowing the return type to `Buffer<ArrayBuffer>` if the input does not include any types that may be `SharedArrayBuffer` if needed.

I'm fine with removing the last two parameters. I don't use them. I added to align with the docs in node (https://nodejs.org/docs/latest/api/buffer.html#static-method-bufferfromobject-offsetorencoding-length).